packages/gateway/src/Shard.ts

```diff
@@ -211,19 +211,51 @@ export class DiscordenoShard {
       }
 
       if (this.gatewayConfig.transportCompression === TransportCompression.zstd) {
-        const fzstd = await getFZStd().catch(() => {
-          this.logger.warn('[Shard] "fzstd" is not installed. Disabled transport compression.')
-          url.searchParams.delete('compress')
+        if ('createZstdDecompress' in zlib) {
+          this.logger.debug('[Shard] Using node:zlib zstd decompression.')
 
-          return null
-        })
+          this.inflateBuffer = null
+          this.inflate = zlib.createZstdDecompress({
+            chunkSize: 64 * 1024,
+          })
+
+          this.inflate.on('error', (e) => {
+            this.logger.error('The was an error in decompressing a Zstd compressed payload', e)
+          })
+
+          this.inflate.on('data', (data) => {
+            if (!(data instanceof Uint8Array)) return
+
+            if (this.inflateBuffer) {
+              const newBuffer = new Uint8Array(this.inflateBuffer.byteLength + data.byteLength)
+              newBuffer.set(this.inflateBuffer)
+              newBuffer.set(data, this.inflateBuffer.byteLength)
+              this.inflateBuffer = newBuffer
 
-        if (fzstd) {
-          this.zstdDecompress = new fzstd.Decompress((data) => {
-            const decodedData = this.textDecoder.decode(data)
-            const parsedData = JSON.parse(decodedData)
-            this.decompressionPromisesQueue.shift()?.(parsedData)
+              return
+            }
+
+            this.inflateBuffer = data
+          })
+        } else {
+          this.logger.debug('[Shard] Attempting to use fzstd zstd decompression.')
+
```

Comment:
I don't think this is necessary, like in mind there should only be three path: log using node:zlib, log using fzstd, or log disabling compression

```suggestion
```